extensions/http-curl/client/HTTPCallback.h

```diff
@@ -33,148 +34,195 @@ namespace minifi {
 namespace utils {
 
 /**
- * will stream as items are processed.
+ * The original class here was deadlock-prone, undocumented and was a smorgasbord of multithreading primitives used inconsistently.
+ * This is a rewrite based on the contract inferred from this class's usage in utils::HTTPClient
+ * through HTTPStream and the non-buggy part of the behaviour of the original class.
+ * Based on these:
```

Comment:
Normally I agree, but this class is pretty poorly designed even in this refactored state, because it had to conform to the existing interface and its current usage by HTTPStream.
Understanding these original limitations is a vital part of understanding why this class is the way it is.
This whole flow should be rethought and rewritten entirely - I planned to create a follow-up issue for it, and I will soon - in the interim, this spares the next people coming along from having to trace back why things are designed this way.